contracts/minter-suite/Minters/MinterSEAV0.sol

```diff
@@ -457,59 +519,59 @@ contract MinterSEAV0 is ReentrancyGuard, MinterBase, IFilteredMinterSEAV0 {
      */
-    function settleAndCreateBid(
+    function settleAuctionAndCreateBid(
         uint256 _settleTokenId,
         uint256 _bidTokenId
     ) external payable {
+        // ensure tokens are in the same project
+        require(
+            _settleTokenId / ONE_MILLION == _bidTokenId / ONE_MILLION,
+            "Only tokens in same project"
+        );
         // settle completed auction, if applicable
-        // @dev in case of frontrun, settleAuction returns early and avoids
-        // reverting
         settleAuction(_settleTokenId);
-        // initialize the auction for the next token
-        createBid_4cM(_bidTokenId);
+        // attempt to bid on next token
+        createBid_l34(_bidTokenId);
     }
 
     /**
-     * @notice Settles a completed auction for `_tokenId`, if one exists.
-     * If there is no active auction for token ID `_tokenId`, or if the auction
-     * has already been settled, this function does not revert, but also does
-     * not modify state (since there is no more required action).
-     * This function reverts if the auction for `_tokenId` exists but has not
+     * @notice Settles a completed auction for `_tokenId`, if it exists and is
+     * not yet settled.
+     * Returns early (does not modify state) if
+     *   - there is no initialized auction for the project
+     *   - there is an auction that has already been settled for the project
+     *   - there is an auction for a different token ID on the project
+     *     (likely due to a front-run)
+     * This function reverts if the auction for `_tokenId` exists, but has not
      * yet ended.
-     * Edge case: if `_tokenId` is nonsense (i.e. not a valid token ID), this
-     * function will not revert, but also will not modify state. This is to
-     * prevent nuisance reverting in the case of a user being front-run when
-     * calling `settleAndCreateBid`.
      * @param _tokenId Token ID to settle auction for.
      */
     function settleAuction(uint256 _tokenId) public nonReentrant {
         uint256 _projectId = _tokenId / ONE_MILLION;
         ProjectConfig storage _projectConfig = projectConfig[_projectId];
         Auction storage _auction = _projectConfig.activeAuction;
         // CHECKS
-        if (_auction.tokenId != _tokenId) {
-            // no active auction for this token
-            return;
-        }
-        if (_auction.settled) {
-            // auction already settled
+        if (
+            (!_auction.initialized) ||
+            _auction.settled ||
+            (_auction.tokenId != _tokenId)
+        ) {
+            // auction not initialized, already settled, or is for a different
+            // token ID, so return early
             return;
         }
         require(block.timestamp > _auction.endTime, "Auction not yet ended");
-        // @dev this check is not strictly necessary, but is included for
-        // clear error messaging
-        require(_auction.initialized, "Auction not initialized");
```

Comment:
What's the rationale for pulling this out of the require (and up above instead)?

Reply:
good callout - seems clearer to revert with this error message. Updated in fc3b08d 🙏